squire_lib/src/ffi_rounds.rs

```diff
@@ -0,0 +1,160 @@
+use std::ptr;
+use uuid::Uuid;
+
+impl RoundId {
+    /// Returns the round if it can be found in the tournament
+    fn get_tourn_round(self, tid: TournamentId) -> Option<Round> {
+        unsafe {
+            match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get(&tid) {
+                Some(t) => {
+                    return t.round_reg.get_round(&self.into()).cloned();
+                }
+                None => {
+                    println!("[FFI]: Cannot find tournament in get_tourn_round();");
+                    return None;
+                }
+            }
+        }
+    }
+
+    /// Gets the round number
+    /// -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_match_number(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.match_number as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Gets the table number
+    /// Warning: Currently cannot detect if the round has not been allocated a table number
+    /// -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_table_number(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.table_number as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Gets the status for a round
+    /// Dead on error
+    #[no_mangle]
+    pub extern "C" fn rid_status(self, tid: TournamentId) -> RoundStatus {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.status;
+            }
+            None => {
+                return RoundStatus::Dead;
+            }
+        }
+    }
+
+    /// Returns the amount of time left in a round
+    /// Retrusn -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_time_left(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.time_left().as_secs() as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Returns the total duration in a round
+    /// Retrusn -1 on error
+    #[no_mangle]
+    pub extern "C" fn rid_duration(self, tid: TournamentId) -> i64 {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                return r.length.as_secs() as i64;
+            }
+            None => {
+                return -1;
+            }
+        }
+    }
+
+    /// Gets the players that are in a round
+    /// Returns NULL on error
+    #[no_mangle]
+    pub extern "C" fn rid_players(self, tid: TournamentId) -> *const PlayerId {
+        match self.get_tourn_round(tid) {
+            Some(r) => {
+                let players: HashSet<PlayerId> = r.players;
+                let len: usize = (players.len() + 1) * std::mem::size_of::<PlayerId>();
+
+                unsafe {
```

Comment:
```suggestion
        let ptr = System
            .allocate(Layout::from_size_align(len, 1).unwrap())
            .unwrap()
            .as_mut_ptr() as *mut PlayerId;

        let slice = unsafe { &mut *(ptr::slice_from_raw_parts(ptr, len) as *mut [PlayerId]) };

        slice.iter_mut().zip(players.iter()).for_each(|(dst, p)| {
            *dst = **p;
        });
        slice[len-1] = Uuid::default().into();

        ptr
```

Reply:
Fixed in f456ca5b640e0d4542d25abbdbd1e399ef53144b